File: src/lib/api/notes.ts
import { invoke } from "@tauri-apps/api/core";
import type { Note } from "./types";

// ── Notes ───────────────────────────────────────────────────────────────

export async function getNote(featureId: string): Promise<Note | null> {
  return invoke<Note | null>("get_note", { featureId });
}

export async function saveNote(featureId: string, content: string): Promise<Note> {
  return invoke<Note>("save_note", { featureId, content });
}

File: src-tauri/src/commands/notes.rs
use tauri::State;

use crate::db;
use crate::AppState;

#[tauri::command]
pub fn get_note(
    state: State<'_, AppState>,
    feature_id: String,
) -> Result<Option<db::notes::Note>, String> {
    let conn = state.db.lock().map_err(|e| e.to_string())?;
    db::notes::get_note(&conn, &feature_id)
}

#[tauri::command]
pub fn save_note(
    state: State<'_, AppState>,
    feature_id: String,
    content: String,
) -> Result<db::notes::Note, String> {
    let conn = state.db.lock().map_err(|e| e.to_string())?;
    db::notes::save_note(&conn, &feature_id, &content)
}
